dynamax/nonlinear_gaussian_ssm/inference_ekf.py

```diff
@@ -128,12 +132,28 @@ def _step(carry, t):
         # Predict the next state
         pred_mean, pred_cov = _predict(filtered_mean, filtered_cov, f, F, Q, u)
 
-        return (ll, pred_mean, pred_cov), (filtered_mean, filtered_cov)
+        # Build carry states
+        carry = {
```

Comment:
I like the idea but the naming is a little confusing. I’d call this dict “outputs” and then only include filtered means and filtered covariances in it.